### PR TITLE
Create admin API to list namespaces

### PR DESCRIPTION
**Pull request by feedmeapples**
**What changed?**

Created an admin API that allows to list namespaces

**Why?**

removing direct DB connection in tctl

**How did you test it?**

tested with `tctl admin namespace list`

**Potential risks**

afaik no added risks, only possible errors with the new API itself

**Is hotfix candidate?**

### CODE REVIEW

File: client/admin/client.go [hunk 1]
```diff
@@ -245,6 +245,20 @@ func (c *clientImpl) GetReplicationMessages(
 	return client.GetReplicationMessages(ctx, request, opts...)
 }
 
+func (c *clientImpl) GetNamespaces(
```
Comment (wxing1292) [inline client/admin/client.go:248]: `GetNamespaces` -> `ListNamespaces`

File: proto/internal/temporal/server/api/adminservice/v1/request_response.proto [hunk 2]
```diff
@@ -37,9 +37,26 @@ import "temporal/server/api/enums/v1/common.proto";
...
-import "temporal/server/api/replication/v1/message.proto";
+
+message GetNamespacesRequest {
+    int32 page_size = 1;
+    bytes next_page_token = 2;
+}
+
+message GetNamespacesResponse {
+    repeated DescribeNamespaceResponse namespaces = 1;
+    bytes next_page_token = 2;
```
Comment (wxing1292) [inline proto/internal/temporal/server/api/adminservice/v1/request_response.proto:52]: APIs using pagination token should be called List*

**Review by wxing1292 — CHANGES_REQUESTED** [review]

File: proto/internal/temporal/server/api/adminservice/v1/request_response.proto (hunk 2 above)
Comment (feedmeapples) [reply proto/internal/temporal/server/api/adminservice/v1/request_response.proto:52]: addressed and also here https://github.com/temporalio/temporal/pull/2159

File: client/admin/client.go (hunk 1 above)
Comment (feedmeapples) [reply client/admin/client.go]: renamed

**Review by wxing1292 — APPROVED** [review]

**Pull request merged**